Allow users to specify a release for a package

awgreene opened the pull request:
Currently, the AppRegistry Server will download the latest release of
a package from quay. This update allows users to specify a specific
release.

awgreene committed:
Allow users to specify a release for a package

Currently, the AppRegistry Server will download the latest release of
a package from quay. This update allows users to specify a specific
release.

tkashem commented on cmd/appregistry-server/main.go:
```diff
@@ -81,19 +82,38 @@ func runCmdFunc(cmd *cobra.Command, args []string) error {
 
 	logger := logrus.WithFields(logrus.Fields{"type": "appregistry", "port": port})
 
+	packageList := strings.Split(packages, ",")
```
Another alternative, a bit more structured
```
type Package struct {
     Name string
     Release string
}
type Input struct {
	// Sources is the set of remote operator source(s) specified where operator
	// manifest(s) are located.
	Sources []*Source

	// Packages is the set of package name(s) specified.
	Packages []Package
}
```

kevinrizza commented on cmd/appregistry-server/main.go:
```diff
@@ -81,19 +82,38 @@ func runCmdFunc(cmd *cobra.Command, args []string) error {
 	logger := logrus.WithFields(logrus.Fields{"type": "appregistry", "port": port})
 
+	packageList := strings.Split(packages, ",")
+	package2release := map[string]string{}
```
This is a messy variable name

awgreene commented:
@tkashem I have addressed your comments, please let me know if there are any additional changes you'd like to see.

awgreene committed:
[bug] Remove duplicate packages from list

This change fixes a bug in the sanatizePackageList function that
allows duplicate packages to be included in the list and adds
a unit test to prevent regression.

openshift-merge-robot committed:
Merge pull request #52 from awgreene/fix-duplicate-package-bug

[bug] Remove duplicate packages from list
